modules/augment/src/gaussianblur.cpp

```diff
@@ -0,0 +1,39 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "precomp.hpp"
+namespace cv { namespace augment {
+
+GaussianBlur::GaussianBlur(int minKernelSize, int maxKernelSize, float minSigma, float maxSigma)
+{
+    this->minKernelSize = minKernelSize;
+    this->maxKernelSize = maxKernelSize;
+    this->minSigma = minSigma;
+    this->maxSigma = maxSigma;
+}
+
+GaussianBlur::GaussianBlur(int minKernelSize, int maxKernelSize)
+{
+    this->minKernelSize = minKernelSize;
+    this->maxKernelSize = maxKernelSize;
+    minSigma = maxSigma = 0; //the sigma will be decided depending on the kernel size
+}
+
+void GaussianBlur::init(const Mat& image)
+{
+    currentKernelSize = Transform::rng.uniform(minKernelSize, maxKernelSize+1); //+1 because the RNG.uniform excludes the max number
+    if (currentKernelSize % 2 == 0)
+    {
+        if (currentKernelSize > minKernelSize) currentKernelSize--;
+        else if (currentKernelSize < maxKernelSize) currentKernelSize++;
+    }
```

Comment:
We need to guarantee that kernel size is odd. Try this random sampling:

```cpp
kernelSize = 2 * rand(min / 2, max / 2 + max % 2) + 1
```

| min | max | possible out values |
|---|----|---|
| 1 | 7 |  1 3 5 7 | 
| 2 6 |  1 2 | 3 5 |
| 2 7 |  1 2 3 |  3 5 7 |
| 3 6 |  1 2 |  3 5 |